### PR TITLE
Added auto detection to .mzXML files

### PR DESCRIPTION
**Pull request by Mailaender**
This is a followup of https://github.com/eclipse/chemclipse/pull/534 which brings it in line with the other recent XML file formats

### CODE REVIEW

File: chemclipse/plugins/org.eclipse.chemclipse.msd.converter.supplier.mzxml/src/org/eclipse/chemclipse/msd/converter/supplier/mzxml/converter/MassSpectrumMagicNumberMatcher.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			}
+			DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+			DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+			Document document = documentBuilder.parse(file);
+			NodeList nodeList = document.getElementsByTagName(IConstants.NODE_MS_RUN);
+			//
+			JAXBContext jaxbContext = JAXBContext.newInstance(IFormat.CONTEXT_PATH_V_320);
+			Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+			MsRun msRun = (MsRun)unmarshaller.unmarshal(nodeList.item(0));
+			if(msRun.getScan().size() == 1)
```
Comment (eselmeister) [inline chemclipse/plugins/org.eclipse.chemclipse.msd.converter.supplier.mzxml/src/org/eclipse/chemclipse/msd/converter/supplier/mzxml/converter/MassSpectrumMagicNumberMatcher.java]: Please put the if condition in curly brackets {}.

File: chemclipse/plugins/org.eclipse.chemclipse.msd.converter.supplier.mzxml/src/org/eclipse/chemclipse/msd/converter/supplier/mzxml/converter/ChromatogramMagicNumberMatcher.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			}
+			DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+			DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+			Document document = documentBuilder.parse(file);
+			NodeList nodeList = document.getElementsByTagName(IConstants.NODE_MS_RUN);
+			//
+			JAXBContext jaxbContext = JAXBContext.newInstance(IFormat.CONTEXT_PATH_V_320);
+			Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+			MsRun msRun = (MsRun)unmarshaller.unmarshal(nodeList.item(0));
+			if(msRun.getScan().size() > 1)
```
Comment (eselmeister) [inline chemclipse/plugins/org.eclipse.chemclipse.msd.converter.supplier.mzxml/src/org/eclipse/chemclipse/msd/converter/supplier/mzxml/converter/ChromatogramMagicNumberMatcher.java]: Please put the if condition in curly brackets {}.

**Comment by eselmeister** [comment]
Please have a look at the comment. Principally, it looks good, but having a deep look at the *.xml content could lead to performance issues. But I don't see any other way to precisely check the file.

**Comment by Mailaender** [comment]
Yes, I also felt bad about it, so I added https://github.com/eclipse/chemclipse/pull/655 to investigate. Rebased and the cost is 4 ms per file on local SSD:
> INFO  main converter.AbstractSupplierFileIdentifier.isMatchMagicNumber (AbstractSupplierFileIdentifier.java:173) Magic number check of mabr_eqa20qty_0006_2L1.mzXML by mzXML Chromatogram (*.mzXML) took 0.002 seconds.
INFO  main converter.AbstractSupplierFileIdentifier.isMatchMagicNumber (AbstractSupplierFileIdentifier.java:173) Magic number check of mabr_eqa20qty_0006_2L1.mzXML by mzXML Mass Spectra (*.mzXML) took 0.002 seconds.

which is noticeable when loading folders with >100 files e.g. https://osf.io/d6hzx/

**Pull request merged**